zio-http/src/main/scala/zhttp/service/client/ClientInboundHandler.scala

```diff
@@ -28,15 +28,15 @@ final class ClientInboundHandler[R](
   override def channelRead0(ctx: ChannelHandlerContext, msg: FullHttpResponse): Unit = {
     msg.touch("handlers.ClientInboundHandler-channelRead0")
 
-    zExec.unsafeRun(ctx)(promise.succeed(ClientResponse.unsafeFromJResponse(msg)))
+    zExec.unsafeRun(ctx)(promise.succeed(ClientResponse.unsafeFromJResponse(msg)).uninterruptible)
     if (isWebSocket) {
       ctx.fireChannelRead(msg.retain())
       ctx.pipeline().remove(ctx.name()): Unit
     }
   }
 
   override def exceptionCaught(ctx: ChannelHandlerContext, error: Throwable): Unit = {
-    zExec.unsafeRun(ctx)(promise.fail(error))
+    zExec.unsafeRun(ctx)(promise.fail(error).uninterruptible)
```

Comment:
Is it possible to write a test case for this?

Reply:
If it's hard to test then we should write a note about what is the issue and why are we making it interruptible.

Reply:
Added note as the test quite hard to be written.